### PR TITLE
Track repo activation on the client side

### PR DESCRIPTION
**Pull request by jcmorrow**
In order to get correct statistics for our paid marketing efforts we
need conversion events to be tracked on the client side so that segment
can drop in the appropriate conversion pixels.

https://trello.com/c/sublWZZi

### CODE REVIEW

File: app/assets/javascripts/directives/repo.js.coffee.erb [hunk 1]
```diff
@@ -5,7 +5,7 @@ App.directive 'repo', ['Subscription', 'StripeCheckout', 'User', (Subscription,
     activateRepo = ->
       scope.processing = true
 
-      scope.repo.$activate()
+      scope.repo.$activate().then((repo) -> track_repo_activated(repo))
         .catch((response) -> alert(errorMessage(response.data)))
```
Comment (jcmorrow) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: Coffeescript's implicit return interacts weirdly with promise chaining sometimes. It's hard for me to tell if this `catch` will still work appropriately in all cases.

**Comment by jcmorrow** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;sublWZZi&#x2F;760-move-the-hound-payment-analytics-event-client-side"> Move the Hound payment analytics event client-side</a>

File: app/assets/javascripts/directives/repo.js.coffee.erb (hunk 1 above)
Comment (gylaz) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: Yes, `catch` should fire for all failures (even those that can happen in `then`'s success).

What do you think about bringing `.then` on a new line, like `catch`?

**Comment by gylaz** [comment]
What is the value of moving this event tracking from server-side to client-side? /cc @croaky

File: app/assets/javascripts/directives/repo.js.coffee.erb [hunk 2]
```diff
@@ -84,4 +85,14 @@ App.directive 'repo', ['Subscription', 'StripeCheckout', 'User', (Subscription,
...
       else
         createSubscription()
+
+    track_repo_activated = (repo) ->
+      window.analytics.track(
+        "Repo Activated",
+        properties: {
+          name: repo.full_github_name,
+          private: repo.private,
+          revenue: repo.plan_price
```
Comment (thorncp) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: Is `plan_price` available here? It looks like the API renders this as [`price_in_cents`](https://github.com/houndci/hound/blob/78684b7d4ae0285cb168bc5f0cb80e64b1acf9b3/app/serializers/repo_serializer.rb#L10).

**Comment by croaky** [comment]
> What is the value of moving this event tracking from server-side to client-side? /cc @croaky

In the current state, we can't tell which channels (Twitter, Facebook, Google) or specific campaigns are performing well or poorly. This change fixes that, which enables us to iterate on paid marketing experiments, which should drive new users getting value from Hound.

**Comment by croaky** [comment]
https://help.segment.com/hc/en-us/articles/205511905-Client-side-vs-server-side-event-tracking

File: app/assets/javascripts/directives/repo.js.coffee.erb (hunk 1 above)
Comment (jcmorrow) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: I like the newline idea. I had it like that originally and moved it  to the same line to be consistent with line 56, but I'm going to change it back.

File: app/assets/javascripts/directives/repo.js.coffee.erb (hunk 2 above)
Comment (jcmorrow) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: Good catch! I've addressed this in10dfa330495102ea33ae91ff351d73703ac10851

File: app/assets/javascripts/directives/repo.js.coffee.erb [hunk 3]
```diff
@@ -84,4 +86,15 @@ App.directive 'repo', ['Subscription', 'StripeCheckout', 'User', (Subscription,
           deactivateRepo()
       else
         createSubscription()
+
+    track_repo_activated = (repo) ->
+      console.log(repo);
```
Comment (croaky) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: Would be cool if Hound could catch leftover debugging statements

Comment (jcmorrow) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: I support this idea!

File: app/assets/javascripts/directives/repo.js.coffee.erb [hunk 4]
```diff
@@ -84,4 +86,14 @@ App.directive 'repo', ['Subscription', 'StripeCheckout', 'User', (Subscription,
...
       else
         createSubscription()
+
+    track_repo_activated = (repo) ->
+      window.analytics.track(
+        "Repo Activated",
+        properties: {
+          name: repo.full_github_name,
+          private: repo.private,
+          revenue: (repo.price_in_cents / 100).toFixed(2)
```
Comment (croaky) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: Might be nice to have a `repo.price_in_dollars` method but I don't feel strongly.

**Comment by croaky** [comment]
I don't know much about Angular so if you wanted a second pair of eyes there, feel free. Code looks good to me!

File: app/assets/javascripts/directives/repo.js.coffee.erb (hunk 3 above)
Comment (gylaz) [inline app/assets/javascripts/directives/repo.js.coffee.erb]: Hmm, I have a vague recollection of CoffeeLint doing that. If so, I wonder why it didn't catch this.

**Comment by jcmorrow** [comment]
I really wanted to add a JS spec for this, but it feels like a lot of work given how little I know of angular/jasmine. I did verify that the events are working via segment. What do you think @gylaz?

**Comment by gylaz** [comment]
> I really wanted to add a JS spec for this, but it feels like a lot of work given how little I know of angular/jasmine. I did verify that the events are working via segment.
👍 We only have 1 or 2 JS tests anyways 😞

**Comment by gylaz** [comment]
:shipit:

**Pull request merged**